drivers/ethernet/eth_adin2111.c

```diff
@@ -320,6 +637,22 @@ static void adin2111_offload_thread(void *p1, void *p2, void *p3)
 			adin2111_port_on_phyint(ctx->port[1]);
 		}
 
+		if (ctx->oa) {
+			if (status1 & ADIN2111_STATUS1_P1_RX_RDY) {
+				ret = eth_adin2111_oa_data_read(dev, 0);
+				if (ret < 0) {
+					break;
+				}
+			}
+			if (status1 & ADIN2111_STATUS1_P2_RX_RDY) {
+				ret = eth_adin2111_oa_data_read(dev, 1);
+				if (ret < 0) {
+					break;
+				}
+			}
+			goto continue_unlock;
+		}
```

Comment:
@spectrum70 , completely missed during the review, there should be no `break` for the `ctx->oa` code; as there is no `while`. The `break` will terminate the irq offload thread. Because the thread is `K_ESSENTIAL` the kernel should crash.